app/src/main/kotlin/com/wire/android/notification/WireNotificationManager.kt

```diff
@@ -336,20 +338,26 @@ class WireNotificationManager @Inject constructor(
     ) {
         appLogger.d("$TAG observe incoming calls")
 
-        coreLogic.getSessionScope(userId).observeE2EIRequired()
-            .map { it is E2EIRequiredResult.NoGracePeriod }
-            .distinctUntilChanged()
-            .flatMapLatest { isBlockedByE2EIRequired ->
-                if (isBlockedByE2EIRequired) {
-                    appLogger.d("$TAG calls were blocked as E2EI is required")
-                    flowOf(listOf())
-                } else {
-                    coreLogic.getSessionScope(userId).calls.getIncomingCalls()
+        coreLogic.getSessionScope(userId).let { userSessionScope ->
+            userSessionScope.observeE2EIRequired()
+                .map { it is E2EIRequiredResult.NoGracePeriod }
+                .distinctUntilChanged()
+                .flatMapLatest { isBlockedByE2EIRequired ->
+                    if (isBlockedByE2EIRequired) {
+                        appLogger.d("$TAG calls were blocked as E2EI is required")
+                        flowOf(listOf())
+                    } else {
+                        userSessionScope.calls.getIncomingCalls()
+                    }.map { calls ->
+                        userSessionScope.users.getSelfUser().first().let {
+                            it.handle ?: it.name ?: ""
```

Comment:
When handle and/or name can be null? If there is no such "normal" case we such at least log this as error if not crash the app to prevent strange app behaviour

Reply:
I don't know if this actually can happen but it's possible as in the database `User` table both `name` and `handle` are not marked as `NOT NULL`, so here it's just done this way to make sure it won't break the app and in my opinion it's better for the user to show him empty user name than to crash the entire app. 
By looking at `UserProfileDTO`, probably at least `name` could be `NOT NULL` but it should be changed in the kalium starting with updating the `User` table in the database.
For now, I can add a log to gather this information and check if there is even such a case at all.